openmdao/utils/options_dictionary.py

```diff
@@ -44,6 +44,23 @@ def __init__(self, parent_name=None, read_only=False):
         self._parent_name = parent_name
         self._read_only = read_only
 
+    def __getstate__(self):
+        """
+        Return state as a dict.
+
+        Returns
+        -------
+        dict
+            State to get.
+        """
+        state = self.__dict__.copy()
+        user_options = state['_dict'].copy()
+        for i in user_options.copy():
+            if not user_options[i]['recordable']:
+                del user_options[i]
+
+        return user_options
```

Comment:
I would add an attribute to `OptionsDictionary` called `_all_recordable` or something like that.  Default it to True and set it to False if `recordable` ever equals False in `declare`.  Then in your `__getstate__`, if `_all_recordable` is True you can just skip all of the looping and just return `self.__dict__`.